packages/headless/src/utils/facet-utils.ts

```diff
@@ -5,7 +5,14 @@ export function sortFacets<T extends {facetId: string}>(
   const payloadMap: Record<string, T> = {};
   facets.forEach((f) => (payloadMap[f.facetId] = f));
 
-  return sortOrder
-    .map((id) => payloadMap[id])
-    .filter((payload) => payload !== undefined);
+  const sortedFacets: T[] = [];
+  sortOrder.forEach((facetId) => {
+    if (facetId in payloadMap) {
+      sortedFacets.push(payloadMap[facetId]);
+      delete payloadMap[facetId];
+    }
+  });
+  const remainingFacets = Object.values(payloadMap);
```

Comment:
In theory, `Object.values` should return an array of the remaining facets in whichever random order, which I don't think is really an issue since those are meant to be the unsorted facets.

In practice, I've noticed that `Object.values` returns values in the order they were added, but I haven't tested this since, again, I don't think the order being random for those unsorted values matters.

Reply:
Indeed, I've read that object insertion order should not be relied on, but I'm not sure under what circumstances the order could differ.

One way to guarantee order of non-existing facets is preserved could be to get them from the original array instead of the payload map object. It's not much longer,

```
const remainingFacets = facets.filter(facet=> !payloadMap[facet.id]))
```